source/src/GeometryUtil.cc

```diff
@@ -57,3 +57,43 @@ dd4hep::rec::LayeredCalorimeterData const* MarlinUtil::getLayeredCalorimeterData
 
   return theExtension;
 }
+
+
+dd4hep::rec::ZPlanarData* MarlinUtil::getVXDData(){
+    dd4hep::Detector& theDetector = dd4hep::Detector::getInstance();
+    dd4hep::DetElement vxdDet = theDetector.detector("VXD");
+    dd4hep::rec::ZPlanarData* vxd = vxdDet.extension <dd4hep::rec::ZPlanarData>();
+    return vxd;
+}
```

Comment:
```suggestion
template<typename DetData>
DetData* MarlinUtil::getDetectorExtensionData(const std::string& name) {
  auto& theDetector = dd4hep::Detector::getInstance();
  auto det = theDetector.detector(name);
  auto* detData = det.extension<DetData>();
  return detData;
}

dd4hep::rec::ZPlanarData* MarlinUtil::getVXDData() {
    return getDetectorExtensionData<dd4hep::rec::ZPlanarData>("VXD");
}
```
And similar for the other implementations, as they are really always the same function that only differ by the extension data they return and by the name of the detector.

Note: You will have to define the template function in the header file, but you can call it in the source file.